### PR TITLE
(FMT): fix spacing in method calls

### LINKED ISSUE #526: Fix spacing in method calls
Test case:

```
fn main() {
    foo::<i32>       ();
    foo.foo::<i32>   ();
}
```

Goal:

```
fn main() {
    foo::<i32>();
    foo.foo::<i32>();
}
```

Current formatting: 

```
fn main() {
    foo::<i32>();
    foo.foo::<i32>   ();
}
```

cc @jajakobyly

### PR DESCRIPTION
**Pull request by jajakobyly**
Fixes #526

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/formatter/impl/spacing.kt [hunk 1]
```diff
@@ -83,6 +83,7 @@ fun createSpacingBuilder(commonSettings: CommonCodeStyleSettings, rustSettings:
         .between(IDENTIFIER, GENERIC_ARGS).spaceIf(false)
         .between(IDENTIFIER, ARG_LIST).spaceIf(false)
         .between(GENERIC_PARAMS, PARAMS_LIKE).spaceIf(false)
+        .between(GENERIC_ARGS, ARG_LIST).spaceIf(false)
         .beforeInside(ARG_LIST, CALL_EXPR).spaceIf(false)
```
Comment (matklad) [inline src/main/kotlin/org/rust/ide/formatter/impl/spacing.kt]: Can both method_call and call_expr be handled by the same rule?

Comment (jajakobyly) [inline src/main/kotlin/org/rust/ide/formatter/impl/spacing.kt]: Unfortunately, no. These guys from test have totally different PSI.

`SpacingBuilder`'s rules apply only to neighbouring siblings in the AST. That's why this didn't work for method call's.
idea_2016-07-14_21-01-02

Comment (matklad) [inline src/main/kotlin/org/rust/ide/formatter/impl/spacing.kt]: Something like

```
      .before(ARG_LIST).spaceIf(false)
```

perhaps?

Comment (jajakobyly) [inline src/main/kotlin/org/rust/ide/formatter/impl/spacing.kt]: yeah :sweat_smile:

**Comment by matklad** [comment]
Thanks!

**Pull request merged**